crypto/ex_data.c

```diff
@@ -287,7 +287,7 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
+    if (!CRYPTO_set_ex_data(to, mx - 1, CRYPTO_get_ex_data(to, mx - 1)))
```

Comment:
Suppose my ex_data were a `malloc`ed string. Then my `dup_func` is probably `strdup`.  If this succeeds but `dup_func` fails, I'll end up with both `to` and `from` believing they own the same pointer and double-free. Or am I confused?